### PR TITLE
feat(specs): add CSV source

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

Add CSV Source for Data Ingestion client API.

### Changes included:

Add CSV to Source type.

### CODE REVIEW

**Review by Fluf22 — DISMISSED** [review]
Seems 👌🏻

File: specs/ingestion/common/schemas/source.yml [hunk 1]
```diff
@@ -146,11 +146,29 @@ SourceJson:
   required:
     - url
 
+SourceCsv:
+  type: object
+  additionalProperties: false
+  properties:
+    url:
+      type: string
```
Comment (millotp) [inline specs/ingestion/common/schemas/source.yml:154]: Can you add description to the fields that are not obvious pls ?

File: specs/ingestion/common/schemas/source.yml [hunk 2]
```diff
@@ -146,11 +146,29 @@ SourceJson:
...
+      $ref: '#/MethodType'
+  required:
+    - url
+
 SourceInput:
   oneOf:
     - $ref: '#/SourceCommercetools'
     - $ref: '#/SourceBigCommerce'
     - $ref: '#/SourceJson'
+    - $ref: '#/SourceCsv'
```
Comment (millotp) [inline specs/ingestion/common/schemas/source.yml:171]: Maybe we should rename them to SourceJSON and SourceCSV 🤔

**Review by millotp — COMMENTED** [review]
Thanks !

File: specs/ingestion/common/schemas/source.yml (hunk 1 above)
Comment (shortcuts) [reply specs/ingestion/common/schemas/source.yml:158]: ^this

Comment (morganleroi) [reply specs/ingestion/common/schemas/source.yml:158]: yes, done

File: specs/ingestion/common/schemas/source.yml (hunk 2 above)
Comment (morganleroi) [reply specs/ingestion/common/schemas/source.yml]: done

File: specs/ingestion/common/schemas/source.yml [hunk 3]
```diff
@@ -146,11 +146,37 @@ SourceJson:
...
+    url:
+      type: string
+      description: The URL of the file.
+    uniqueIDColumn:
+      type: string
+      description: The name of the column that contains the unique ID.
+    mapping:
+      type: object
+      description: >
+        Mapping of type for every column. For example {"myColumn": "boolean, "myOtherColumn": "json"}.
```
Comment (millotp) [inline specs/ingestion/common/schemas/source.yml:166]:
```suggestion
        Mapping of type for every column. For example {"myColumn": "boolean", "myOtherColumn": "json"}.
```

File: specs/ingestion/common/schemas/source.yml [hunk 4]
```diff
@@ -146,11 +146,37 @@ SourceJson:
...
+SourceCSV:
+  type: object
+  additionalProperties: false
+  properties:
+    url:
+      type: string
+      description: The URL of the file.
+    uniqueIDColumn:
+      type: string
+      description: The name of the column that contains the unique ID.
```
Comment (millotp) [inline specs/ingestion/common/schemas/source.yml:162]:
```suggestion
      description: The name of the column that contains the unique ID, used as `objectID` in Algolia.
```

File: specs/ingestion/common/schemas/source.yml [hunk 5]
```diff
@@ -168,13 +194,34 @@ SourceUpdateCommercetools:
...
+
+SourceUpdateCSV:
   type: object
   additionalProperties: false
   properties:
     url:
       type: string
       description: The URL of the file.
+    uniqueIDColumn:
+      type: string
```
Comment (millotp) [inline specs/ingestion/common/schemas/source.yml:217]: ah I didn't saw those yesterday, when you have to duplicate properties it's better to put them in their own object and use ref, sorry

Comment (morganleroi) [reply specs/ingestion/common/schemas/source.yml:217]: ah yeah, of course

**Review by millotp — APPROVED** [review]
GG !

**Comment by morganleroi** [comment]
Could you allow me to merge PR in this repo @millotp  ?

**Pull request merged**